src/main/java/core/basesyntax/operation/OperationStrategyImpl.java

```
@@ -0,0 +1,18 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.model.Operation;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+
+    private final Map<Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<Operation, OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public OperationHandler getOperationHandler(Operation operation) {
+        return operationHandlers.get(operation);
```

Consider adding a check to handle cases where an operation might not have a corresponding handler. You could throw an exception or return a default handler if the operation is not found in the map.